serverless import gcs/azure (#20230)
pingcap/docs: tidb-cloud/import-csv-files-serverless.md

3. On the **Import Data from Amazon S3** page, provide the following information for the source CSV files:

    - **Import File Count**: select **One file** or **Multiple files** as needed.
    - **Included Schema Files**: this field is only visible when importing multiple files. If the source folder contains the target table schemas, select **Yes**. Otherwise, select **No**.
    - **Data Format**: select **CSV**.
    - **File URI** or **Folder URI**:
        - When importing one file, enter the source file URI and name in the following format `s3://[bucket_name]/[data_source_folder]/[file_name].csv`. For example, `s3://sampledata/ingest/TableName.01.csv`.
        - When importing multiple files, enter the source file URI and name in the following format `s3://[bucket_name]/[data_source_folder]/`. For example, `s3://sampledata/ingest/`.
    - **Bucket Access**: you can use either an AWS Role ARN or an AWS access key to access your bucket. For more information, see [Configure Amazon S3 access](/tidb-cloud/serverless-external-storage.md#configure-amazon-s3-access).
        - **AWS Role ARN**: enter the AWS Role ARN value.

>         - **AWS Access Key**: enter the AWS access key ID and AWS secret access key.

Author: I have already informed the front-end engineer to make the changes.

    - `s3://[bucket_name]/[data_source_folder]/my-data*.csv`: all CSV files in the folder starting with `my-data` will be imported into the same target table.

    Note that only `?` and `*` are supported.

    > **Note:**
    >
    > The URI must contain the data source folder.

6. Click **Start Import**.

7. When the import progress shows **Completed**, check the imported tables.

</div>

<div label="Google Cloud">

1. Open the **Import** page for your target cluster.

    1. Log in to the [TiDB Cloud console](https://tidbcloud.com/) and navigate to the [**Clusters**](https://tidbcloud.com/console/clusters) page of your project.

        > **Tip:**
        >
        > If you have multiple projects, you can click <MDSvgIcon name="icon-left-projects" /> in the lower-left corner and switch to another project.

    2. Click the name of your target cluster to go to its overview page, and then click **Import** in the left navigation pane.

2. Select **Import data from Cloud Storage**, and then click **Google Cloud Storage**.

> 3. On the **Import Data from GCS** page, provide the following information for the source CSV files:

Reviewer: ```suggestion
3. On the **Import Data from Google Cloud Storage** page, provide the following information for the source CSV files:
```

    2. Click the name of your target cluster to go to its overview page, and then click **Import** in the left navigation pane.

2. Select **Import data from Cloud Storage**, and then click **Azure Blob Storage**.

3. On the **Import Data from Azure Blob Storage** page, provide the following information for the source CSV files:

    - **Import File Count**: select **One file** or **Multiple files** as needed.
    - **Included Schema Files**: this field is only visible when importing multiple files. If the source folder contains the target table schemas, select **Yes**. Otherwise, select **No**.
    - **Data Format**: select **CSV**.
    - **File URI** or **Folder URI**:
        - When importing one file, enter the source file URI and name in the following format `[azure|https]://[bucket_name]/[data_source_folder]/[file_name].csv`. For example, `[azure|https]://sampledata/ingest/TableName.01.csv`.
        - When importing multiple files, enter the source file URI and name in the following format `[azure|https]://[bucket_name]/[data_source_folder]/`. For example, `[azure|https]://sampledata/ingest/`.

>     - **Bucket Access**: you can use a SAS token to access your bucket. For more information, see [Configure Azure Blob Storage access](/tidb-cloud/serverless-external-storage.md#configure-azure-blob-storage-access).

Reviewer 2: ```suggestion
    - **Bucket Access**: you can use a shared access signature (SAS) token to access your bucket. For more information, see [Configure Azure Blob Storage access](/tidb-cloud/serverless-external-storage.md#configure-azure-blob-storage-access).
```

---

> title: Import CSV Files from Amazon S3, GCS or Azure Blob Storage into TiDB Cloud Serverless 
> summary: Learn how to import CSV files from Amazon S3, GCS or Azure Blob Storage into TiDB Cloud Serverless.

Reviewer 2: ```suggestion
title: Import CSV Files from Amazon S3, GCS, or Azure Blob Storage into TiDB Cloud Serverless 
summary: Learn how to import CSV files from Amazon S3, GCS, or Azure Blob Storage into TiDB Cloud Serverless.
```

---

> # Import CSV Files from Amazon S3, GCS or Azure Blob Storage into TiDB Cloud Serverless

Reviewer 2: ```suggestion
# Import CSV Files from Amazon S3, GCS, or Azure Blob Storage into TiDB Cloud Serverless
```

> This document describes how to import CSV files from Amazon Simple Storage Service (Amazon S3), Google Cloud Storage (GCS) or Azure Blob Storage into TiDB Cloud Serverless.

Reviewer 2: ```suggestion
This document describes how to import CSV files from Amazon Simple Storage Service (Amazon S3), Google Cloud Storage (GCS), or Azure Blob Storage into TiDB Cloud Serverless.
```

    > **Note:**
    >
    > - To achieve better performance, it is recommended to limit the size of each compressed file to 100 MiB.
    > - The Snappy compressed file must be in the [official Snappy format](https://github.com/google/snappy). Other variants of Snappy compression are not supported.
    > - For uncompressed files, if you cannot update the CSV filenames according to the preceding rules in some cases (for example, the CSV file links are also used by your other programs), you can keep the filenames unchanged and use the **Mapping Settings** in [Step 4](#step-4-import-csv-files-to-tidb-cloud-serverless) to import your source data to a single target table.

## Step 2. Create the target table schemas

Because CSV files do not contain schema information, before importing data from CSV files into TiDB Cloud Serverless, you need to create the table schemas using either of the following methods:

- Method 1: In TiDB Cloud Serverless, create the target databases and tables for your source data.

> - Method 2: In the Amazon S3, GCS or Azure Blob Storage directory where the CSV files are located, create the target table schema files for your source data as follows:

Reviewer 2: ```suggestion
- Method 2: In the Amazon S3, GCS, or Azure Blob Storage directory where the CSV files are located, create the target table schema files for your source data as follows:
```

        ```

    2. Create table schema files for your source data.

        If you do not include the table schema files in the Amazon S3, GCS or Azure Blob Storage directory where the CSV files are located, TiDB Cloud Serverless will not create the corresponding tables for you when you import the data.

        Each table schema file must be in the `${db_name}.${table_name}-schema.sql` format and contain a `CREATE TABLE` DDL statement. With this file, TiDB Cloud Serverless will create the `${db_table}` table in the `${db_name}` database when you import the data.

        For example, if you create a `mydb.mytable-schema.sql` file that contains the following statement, TiDB Cloud Serverless will create the `mytable` table in the `mydb` database when you import the data.

        {{< copyable "sql" >}}

        ```sql
        CREATE TABLE mytable (
        ID INT,
        REGION VARCHAR(20),
        COUNT INT );
        ```

        > **Note:**
        >
        > Each `${db_name}.${table_name}-schema.sql` file should only contain a single DDL statement. If the file contains multiple DDL statements, only the first one takes effect.

## Step 3. Configure cross-account access

> To allow TiDB Cloud Serverless to access the CSV files in the Amazon S3 or GCS bucket, do one of the following:

Reviewer 2: ```suggestion
To allow TiDB Cloud Serverless to access the CSV files in the Amazon S3, GCS, or Azure Blob Storage bucket, do one of the following:
```
